homeassistant/components/caldav/config_flow.py

```diff
@@ -0,0 +1,170 @@
+            )
+            self._abort_if_unique_id_configured()
+
+            if user_input.get(CONF_ADD_CUSTO_CALENDAR):
+                return await self.async_step_custom_calendars(user_input)
+
+            try:
+                await async_caldav_connect(self.hass, user_input)
+                return self.async_create_entry(
+                    title=DOMAIN,
+                    data=user_input,
+                    options={
+                        CONF_CALENDARS: user_input.pop(CONF_CALENDARS, []),
+                        CONF_CUSTOM_CALENDARS: user_input.pop(
+                            CONF_CUSTOM_CALENDARS, []
+                        ),
+                    },
+                )
+            except ConfigEntryNotReady as error:
+                _LOGGER.error(error)
+                errors["base"] = "cannot_connect"
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_custom_calendars(self, user_input=None) -> FlowResult:
+        """Add custom calendars."""
+        if user_input.pop(CONF_ADD_CUSTO_CALENDAR, None):
+            self.user_input = user_input
+        elif user_input:
+            self.user_input.update(user_input)
+            return await self.async_step_user(self.user_input)
+
+        return self.async_show_form(
+            step_id="custom_calendars", data_schema=CUSTOM_CALENDARS_SCHEMA
+        )
+
+    async def async_step_import(self, import_config: dict[str, str]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
+
+
+class CaldavOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle option."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Init object."""
+        self.config_entry = config_entry
+        self.calendars = config_entry.options.get(CONF_CALENDARS, [])
+        self.custom_calendars = config_entry.options.get(CONF_CUSTOM_CALENDARS, [])
+
+    async def async_step_init(self, user_input=None) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors = {}
+        options_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_CALENDARS, default=self.calendars
+                ): selector.SelectSelector(
+                    selector.SelectSelectorConfig(
+                        options=[], custom_value=True, multiple=True
+                    ),
+                ),
+                vol.Optional(
+                    CONF_CUSTOM_CALENDARS,
+                    default=[],
+                    description={"suggested_value": self.custom_calendars},
+                ): selector.ObjectSelector(),
+            }
+        )
+        if user_input:
+            if (
+                len(user_input[CONF_CALENDARS]) > 0
+                and len(user_input[CONF_CUSTOM_CALENDARS]) > 0
```

Comment:
```suggestion
                user_input[CONF_CALENDARS] and user_input[CONF_CUSTOM_CALENDARS]
```

Non-empty lists are truthy in python.